crates/router/src/connector/stripe/transformers.rs

```diff
@@ -2446,7 +2446,7 @@ impl<F> TryFrom<&types::RefundsRouterData<F>> for RefundRequest {
         Ok(Self {
             amount: Some(amount),
             payment_intent,
-            metadata_order_id: item.payment_id.clone(),
+            metadata_order_id: item.request.refund_id.clone(),
```

Comment:
This is a bug fix, But this will break the web hooks since all the older refund will have payment_id in the Metadata.

We can handle this by introducing a new value in metadata, let's say is_refund_id_as_reference: bool from which we can identify whether we have to return connector_refund_id (or) refund_id from this metadata in the web hooks flow